azure-ts-webserver does not build

### What happened?

Trying to build the azure-ts-webserver exmple does not build due to type errors when accessing the array virtualNetwork.subnets. Full error is:

index.ts(37,23): error TS7053: Element implicitly has an 'any' type because expression of type '0' can't be used to index type 'Output<SubnetResponse[] | undefined>'.
      Property '0' does not exist on type 'Output<SubnetResponse[] | undefined>'.

### Steps to reproduce

Download example and try to run "pulumi up"

### Expected Behavior

The server would build

### Actual Behavior

The script errors

### Versions used

3.32.1

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/azure-ts-webserver/index.ts
+++ b/azure-ts-webserver/index.ts
@@ -24,6 +24,9 @@ const virtualNetwork = new network.VirtualNetwork("server-network", {
     }],
 });
 
+// We need to retrieve the subnet before proceeding
+const subnet = virtualNetwork.subnets.apply(subnet => subnet![0])
+
 // Now allocate a public IP and assign it to our NIC.
 const publicIp = new network.PublicIPAddress("server-ip", {
     resourceGroupName,
@@ -34,7 +37,7 @@ const networkInterface = new network.NetworkInterface("server-nic", {
     resourceGroupName,
     ipConfigurations: [{
         name: "webserveripcfg",
-        subnet: { id: virtualNetwork.subnets[0].id },
+        subnet: subnet,
         privateIPAllocationMethod: network.IPAllocationMethod.Dynamic,
         publicIPAddress: { id: publicIp.id },
     }],

--- a/azure-ts-webserver/index.ts
+++ b/azure-ts-webserver/index.ts
@@ -24,9 +24,6 @@ const virtualNetwork = new network.VirtualNetwork("server-network", {
     }],
 });
 
-// We need to retrieve the subnet before proceeding
-const subnet = virtualNetwork.subnets.apply(subnet => subnet![0])
-
 // Now allocate a public IP and assign it to our NIC.
 const publicIp = new network.PublicIPAddress("server-ip", {
     resourceGroupName,
@@ -37,7 +34,7 @@ const networkInterface = new network.NetworkInterface("server-nic", {
     resourceGroupName,
     ipConfigurations: [{
         name: "webserveripcfg",
-        subnet: subnet,
+        subnet: virtualNetwork.subnets.apply(subnet => subnet![0]),
         privateIPAllocationMethod: network.IPAllocationMethod.Dynamic,
         publicIPAddress: { id: publicIp.id },
     }],

--- a/azure-ts-webserver/tsconfig.json
+++ b/azure-ts-webserver/tsconfig.json
@@ -1,6 +1,7 @@
 {
     "compilerOptions": {
         "outDir": "bin",
+        "strict": true,
         "target": "es2016",
         "module": "commonjs",
         "moduleResolution": "node",
@@ -8,9 +9,9 @@
         "experimentalDecorators": true,
         "pretty": true,
         "noFallthroughCasesInSwitch": true,
-        "forceConsistentCasingInFileNames": true,
+        "forceConsistentCasingInFileNames": true
     },
     "files": [
-        "index.ts",
+        "index.ts"
     ]
 }

